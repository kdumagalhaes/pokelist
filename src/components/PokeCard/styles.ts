import styled from 'styled-components'

export const Container = styled.a`
  height: 265px;
  width: 100%;
  max-width: 200px;
  outline: transparent;
  border-radius: 5px;
  transition: outline 0.3s;
  padding: 2rem;
  max-width: 26rem;
  text-decoration: none;
  box-shadow: 0 25px 50px -12px rgb(0 0 0 / 0.25);
  transition: box-shadow 0.4s;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: space-between;

  &:hover {
    box-shadow: 0 20px 25px -5px rgb(0 0 0 / 0.1),
      0 8px 10px -6px rgb(0 0 0 / 0.1);
  }
`
