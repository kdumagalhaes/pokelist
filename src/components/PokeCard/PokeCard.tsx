import { Container } from './styles'
import { NavLink } from 'react-router-dom'
import { capitalizeFirstLetter } from '../../utils/capitalizeFirstLetter'
interface PokeCardProps {
  name: string
  image: string
}

export function PokeCard({ name, image }: PokeCardProps) {
  return (
    <Container>
      <NavLink className="pokemon-link" to={{ pathname: `/pokemon/${name}` }}>
        <img className="pokemon-image" src={image} alt={name} />
        <h2 className="pokemon-name">{capitalizeFirstLetter(name)}</h2>
      </NavLink>
    </Container>
  )
}
