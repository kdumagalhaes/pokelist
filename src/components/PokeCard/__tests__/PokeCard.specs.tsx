import { render, screen } from '@testing-library/react'
import { BrowserRouter } from 'react-router-dom'
import { PokeCard } from '../PokeCard'

describe('PokeCard', () => {
  it('initial component should match snapshot', () => {
    const { asFragment } = render(
      <BrowserRouter>
        <PokeCard name={''} image={''} />
      </BrowserRouter>,
    )
    expect(asFragment()).toMatchSnapshot()
  })

  it('should render pokémon picture', async () => {
    await render(
      <BrowserRouter>
        <PokeCard
          name={'pikachu'}
          image={
            'https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/official-artwork/1.png'
          }
        />
      </BrowserRouter>,
    )
    const image = screen.getByAltText('pikachu')
    const url =
      'https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/official-artwork/1.png'
    expect(image).toHaveAttribute('src', url)
  })

  it('should render pokémon name', () => {
    render(
      <BrowserRouter>
        <PokeCard name={'Pikachu'} image={''} />
      </BrowserRouter>,
    )
    const pokemonName = /pikachu/i
    expect(screen.getByText(pokemonName)).toBeInTheDocument()
  })

  it('should navigate to pokémon details page', () => {
    render(
      <BrowserRouter>
        <PokeCard name={''} image={''} />
      </BrowserRouter>,
    )
    const path = '/pokemon/'
    expect(screen.getByRole('link')).toHaveAttribute('href', path)
  })
})
