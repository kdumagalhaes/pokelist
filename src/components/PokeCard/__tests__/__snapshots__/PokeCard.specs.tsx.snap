// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokeCard initial component should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-bcXHqe bgIxFf"
  >
    <a
      class="pokemon-link"
      href="/pokemon/"
    >
      <img
        alt=""
        class="pokemon-image"
        src=""
      />
      <h2
        class="pokemon-name"
      />
    </a>
  </div>
</DocumentFragment>
`;
